### PR TITLE
Build ObjC examples in travis

### PR DESCRIPTION
**Pull request by y-zeng**
Part of #3371

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,31 @@ env:
     - CONFIG=opt
     - TEST=objc
     - JOBS=1
+  matrix:
+    - TEST_PATH="src/objective-c/tests" WORKSPACE="Tests.xcworkspace"
+      SCHEME="RxLibraryUnitTests" BUILD_ONLY="false" INTEROP_SERVER="false"
```
Comment (jcanizales) [inline .travis.yml]: nit: It'd be nice if you reordered the variables so `SCHEME` is first, followed by `WORKSPACE`. That way it's easier to tell which test is which from the Travis page (see https://travis-ci.org/grpc/grpc/builds/140600382 ).

File: .travis.yml [hunk 2]
```diff
@@ -16,20 +41,24 @@ before_install:
...
-xcode_sdk: iphonesimulator9.3
+  - if [ "${INTEROP_SERVER}" = "true" ]; then
+      make interop_server;
+      (bins/$CONFIG/interop_server --port=5050 &);
+      (bins/$CONFIG/interop_server --port=5051 --use_tls &);
+    fi
+script:
+  - if [ "${BUILD_ONLY}" = "true" ]; then
+      xctool -workspace "$TEST_PATH/$WORKSPACE" -scheme "$SCHEME"
+      -sdk iphonesimulator9.3 clean build;
```
Comment (jcanizales) [inline .travis.yml]: What's the `clean` for?

File: .travis.yml [hunk 3]
```diff
@@ -16,20 +41,24 @@ before_install:
...
+    else
+      xctool -workspace "$TEST_PATH/$WORKSPACE" -scheme "$SCHEME"
+      -sdk iphonesimulator9.3 build test;
```
Comment (jcanizales) [inline .travis.yml]: `test` _should_ trigger a build action, and specifying `test` only might make it faster. Did you try that?

File: .travis.yml (hunk 1 above)
Comment (y-zeng) [inline .travis.yml]: Ugh, that makes the page much clearer. Thanks a lot for the review!

File: .travis.yml (hunk 2 above)
Comment (y-zeng) [inline .travis.yml]: Removed. It seems unnecessary to use `clean` in travis.

File: .travis.yml (hunk 3 above)
Comment (y-zeng) [inline .travis.yml]: Removed `build`. I was trying to use the same command as in https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/objective_c.rb#L82

**Comment by y-zeng** [comment]
Jenkins tests are not related. Travis tests passed.

**Pull request merged**